### ISSUE TITLE
what is the important  of the below lines?

### ISSUE THREAD

**Issue by nile649** [rosinality/style-based-gan-pytorch#11]
if mixing and random.random() < 0.9:
                gen_in11, gen_in12, gen_in21, gen_in22 = torch.randn(4, b_size, code_size, device='cuda').chunk(4, 0)
                gen_in1 = [gen_in11.squeeze(0), gen_in12.squeeze(0)]
                gen_in2 = [gen_in21.squeeze(0), gen_in22.squeeze(0)]

            else:
                gen_in1, gen_in2 = torch.randn(2, b_size, code_size, device='cuda').chunk(2, 0)
                gen_in1 = gen_in1.squeeze(0)
                gen_in2 = gen_in2.squeeze(0)

**Comment by rosinality** [comment]
It is mixing regularization in the paper. Mixing regularization makes model robust when more than 1 latent code used for generation. (style mixing.)

**Comment by nile649** [comment]
Thankyou

**Comment by nile649** [comment]
could you also explain what is inject_index and crossover do.

        if len(style) < 2:
            inject_index = [len(self.progression) + 1]

        else:
            inject_index = random.sample(list(range(step)), len(style) - 1)

        crossover = 0

        for i, (conv, to_rgb) in enumerate(zip(self.progression, self.to_rgb)):
            if mixing_range == (-1, -1):
                if crossover < len(inject_index) and i > inject_index[crossover]:
                    crossover = min(crossover + 1, len(style))

                style_step = style[crossover]

**Comment by rosinality** [comment]
To do mixing regularization, you should choice some layers (index), and use secondary latent codes after that layer. inject_index and crossover is for implement this.

**Comment by nile649** [comment]
style_step = style[crossover] gives learned style from latent code, rather layer where one need to mix. 

I am sorry, I am very much stuck on the same line.

**Comment by rosinality** [comment]
Yes, it works by get style codes using index of layers and inject it to that layer.

**Comment by nile649** [comment]
Thank you,
I have been trying to run your's implementation on CIFAR10.

Below are the only change I did, do you think it makes sense?

for j in range(0,epoch):
    iteration = 0
    data_loader = sample_data_cifar(4 * 2 ** step)
    dataset = iter(data_loader)
    pbar = tqdm(range(len(data_loader)*2))
    for i in pbar:
        discriminator.zero_grad()

        alpha = min(1, 0.00002 * iteration)

        if iteration > 49000: #since cifar has 50k images, after every 49k step would +1, go to next epoch
            alpha = 0
            iteration = 0
            step += 1
